server/etcdserver/api/etcdhttp/health_test.go

```diff
@@ -185,6 +190,156 @@ func TestHealthHandler(t *testing.T) {
+			expectStatusCode: http.StatusInternalServerError,
+		},
+		{
+			name:             "/ready/serializable_read ok if CORRUPT alarm is on",
+			alarms:           []*pb.AlarmMember{{MemberID: uint64(0), Alarm: pb.AlarmType_CORRUPT}, {MemberID: uint64(0), Alarm: pb.AlarmType_NOSPACE}},
+			healthCheckURL:   "/readyz/serializable_read",
+			expectStatusCode: http.StatusOK,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mux := http.NewServeMux()
+			s := &fakeHealthServer{}
+			logger := zaptest.NewLogger(t)
+			InstallHealthEndpoints(logger, mux, s)
+			ts := httptest.NewServer(mux)
+			defer ts.Close()
+			// OK before alarms are activated.
+			checkHttpResponse(t, ts, tt.healthCheckURL, http.StatusOK, nil, nil)
+			// Activate the alarms.
+			s.alarms = tt.alarms
+			checkHttpResponse(t, ts, tt.healthCheckURL, tt.expectStatusCode, tt.inResult, tt.notInResult)
+		})
+	}
+}
+
+func TestSerializableReadCheck(t *testing.T) {
+	tests := []healthTestCase{
+		{
+			name:             "Alive even if authentication failed",
+			healthCheckURL:   "/livez",
+			apiError:         auth.ErrUserEmpty,
+			expectStatusCode: http.StatusOK,
+		},
+		{
+			name:             "Alive even if authorization failed",
+			healthCheckURL:   "/livez",
+			apiError:         auth.ErrPermissionDenied,
+			expectStatusCode: http.StatusOK,
+		},
+		{
+			name:             "Not alive if range api is not available",
+			healthCheckURL:   "/livez",
+			apiError:         fmt.Errorf("Unexpected error"),
+			expectStatusCode: http.StatusInternalServerError,
+		},
+		{
+			name:             "Not ready if range api is not available",
+			healthCheckURL:   "/readyz",
+			apiError:         fmt.Errorf("Unexpected error"),
+			expectStatusCode: http.StatusInternalServerError,
+		},
+		{
+			name:             "Bad request if allowlist a non registered check",
+			healthCheckURL:   "/readyz?allowlist=non_exist&verbose",
+			expectStatusCode: http.StatusBadRequest,
+		},
+		{
+			name:             "Cannot specify both allowlist and exclude at the same time",
+			healthCheckURL:   "/readyz?allowlist=data_corruption&exclude=serializable_read",
+			apiError:         fmt.Errorf("Unexpected error"),
+			expectStatusCode: http.StatusBadRequest,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mux := http.NewServeMux()
+			s := &fakeHealthServer{apiError: tt.apiError}
+			logger := zaptest.NewLogger(t)
+			InstallHealthEndpoints(logger, mux, s)
+			ts := httptest.NewServer(mux)
+			defer ts.Close()
+			checkHttpResponse(t, ts, tt.healthCheckURL, tt.expectStatusCode, tt.inResult, tt.notInResult)
+		})
+	}
+}
+
+func checkHttpResponse(t *testing.T, ts *httptest.Server, url string, expectStatusCode int, inResult []string, notInResult []string) {
+	res, err := ts.Client().Do(&http.Request{Method: http.MethodGet, URL: testutil.MustNewURL(t, ts.URL+url)})
+	if err != nil {
+		t.Errorf("fail serve http request %s %v", url, err)
```

Comment:
suggest to use t.Fatal to abort.

Reply:
Not sure that it's related to my chrome cache. The change seems be reverted before this comment.